### PR TITLE
Translate dropdown and placeholder

### PR DESCRIPTION
**Pull request by ger-benjamin**
This is a WIP, open for discussions.

 1. First, I had to add the ``gettext`` provider. Else, It throws an error because it doesn't find a provider for the ``translate`` keyword.

 ~~2. For the **locationchooser**, I had to change how I create the dropdown. ng-options doesn't work with "on-the-flight translation". So I've used a ``ng-repeat`` instead the ``ng-options``. The consequence is that I can't use an object in the "value" of the options. So now, I must use the extent as a String.~~

 3. I've tried my changes in a GeoMapFish. For the **search**, I've saw that the project doesn't add the translation in the "pot" file. I had to add the same string as in the placeholder in a hidden markup in the html file.

### CODE REVIEW

File: contribs/gmf/src/directives/locationchooser.js [hunk 1]
```diff
@@ -29,10 +29,12 @@ gmf.locationchooserDirective = function() {
...
     controllerAs: 'ctrl',
-    template: '<select ' +
-        'ng-model="ctrl.selectedLocation" ' +
-        'ng-options="location.label for location in ::ctrl.locations" ' +
+    template:
+        '<select ng-model="ctrl.selectedLocation" ' +
         'ng-change="ctrl.setLocation()">' +
+        '<option value="{{location.extent}}"' +
+        'ng-repeat="location in ctrl.locations">' +
+        '{{location.label | translate}}</option>' +
```
Comment (elemoine) [inline contribs/gmf/src/directives/locationchooser.js]: If we decide to use the `translate` directive in GMF directive partials then the `gmf` Angular module should depend on the `gettext` module.

File: contribs/gmf/examples/locationchooser.js [hunk 2]
```diff
@@ -43,10 +43,10 @@ app.MainController = function() {
 
 
   /**
-   * @type {gmfx.LocationchooserLocation}
+   * @type {string}
    * @export
    */
-  this.selectedLocation = this.locations[2];
+  this.selectedLocation = '[' + this.locations[2].extent + ']';
```
Comment (elemoine) [inline contribs/gmf/examples/locationchooser.js]: What are these square brackets for?

File: contribs/gmf/examples/search.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ var app = {};
 
 
 /** @type {!angular.Module} **/
-app.module = angular.module('app', ['gmf']);
+app.module = angular.module('app', ['gmf', 'gettext']);
```
Comment (elemoine) [inline contribs/gmf/examples/search.js]: See my other comment regarding `gettext` being a dependency of the `gmf` module.

File: contribs/gmf/examples/locationchooser.js (hunk 2 above)
Comment (elemoine) [inline contribs/gmf/examples/locationchooser.js]: Ok I think I understand. This is related to your comment in the issue description and the first "yellow" note in the Angular documentation for "select": https://docs.angularjs.org/api/ng/directive/select.

Comment (elemoine) [inline contribs/gmf/examples/locationchooser.js]: But there must be a better way that using this (and `JSON.parse`).

**Comment by elemoine** [comment]
> I've tried my changes in a GeoMapFish. For the search, I've saw that the project doesn't add the translation in the "pot" file. I had to add the same string as in the placeholder in a hidden markup in the html file.

It's related to the Makefile not considering the ngeo partials. If we decide that ngeo directives (or gmf directives) include translations that the extraction and translation should probably be done by ngeo itself. I hope you see what I mean.

**Comment by ger-benjamin** [comment]
Yep, I understand. Thanks @elemoine

**Comment by ger-benjamin** [comment]
I don't understand the Travis error. Someone has any clue ?

**Comment by elemoine** [comment]
The build fails because the `check-example.js` PhantomJS script gives errors on the `locationchooser` example. This means that your example won't work when using `gmf.js` on github.io. You can reproduce the problem in your browser by opening http://localhost:3000/.build/examples-hosted/contribs/gmf/locationchooser.html.

**Comment by elemoine** [comment]
The problem may be related to `angular-gettext.js` not being available. Look at what's done for other external libraries in the `Makefile` (d3 for example). You'll need to mimic that for `angular-gettext.js`.

**Comment by ger-benjamin** [comment]
Thanks @elemoine

**Comment by asaunier** [comment]
See also https://github.com/rubenv/angular-gettext/issues/58

**Comment by ger-benjamin** [comment]
Thanks @asaunier but watch out: I've tried with Angular v1.4.7 and that doesn't work. Then I tried with your (older) v1.3.5 and that works... I don't currently know why...

**Comment by ger-benjamin** [comment]
@asaunier Now it works with your solution. With Angular v1.4+ I've should remove the one-time binding on my 'for'.
 - Before: for location in ctrl.locationsfor location in **::ctrl.locations**
 - Now: for location in ctrl.locationsfor location in ctrl.locations

**Comment by asaunier** [comment]
Good news! Thanks :)

**Comment by sbrunner** [comment]
Can this pullrequest be merged ?

**Comment by ger-benjamin** [comment]
For me, yes.

**Comment by sbrunner** [comment]
=> #415 

Looks good, merge

**Pull request merged**